src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -858,35 +849,34 @@ private RuntimeFieldInfo[] PopulateFields(Filter filter)
 
                 private unsafe void PopulateRtFields(Filter filter, RuntimeType declaringType, ref ListBuilder<RuntimeFieldInfo> list)
                 {
-                    IntPtr* pResult = stackalloc IntPtr[64];
-                    int count = 64;
+                    Span<IntPtr> result = stackalloc IntPtr[64];
+                    int count = 0;
 
-                    if (!RuntimeTypeHandle.GetFields(declaringType, pResult, &count))
+                    if (!RuntimeTypeHandle.GetFields(declaringType, ref result, ref count))
                     {
-                        fixed (IntPtr* pBigResult = new IntPtr[count])
-                        {
-                            RuntimeTypeHandle.GetFields(declaringType, pBigResult, &count);
-                            PopulateRtFields(filter, pBigResult, count, declaringType, ref list);
-                        }
+                        result = new IntPtr[count];
+                        bool success = RuntimeTypeHandle.GetFields(declaringType, ref result, ref count);
+                        Debug.Assert(success && result.Length == count);
+                        PopulateRtFields(filter, result, declaringType, ref list);
                     }
                     else if (count > 0)
                     {
-                        PopulateRtFields(filter, pResult, count, declaringType, ref list);
+                        PopulateRtFields(filter, result.Slice(0, count), declaringType, ref list);
                     }
                 }
 
                 private unsafe void PopulateRtFields(Filter filter,
-                    IntPtr* ppFieldHandles, int count, RuntimeType declaringType, ref ListBuilder<RuntimeFieldInfo> list)
+                    Span<IntPtr> ppFieldHandles, RuntimeType declaringType, ref ListBuilder<RuntimeFieldInfo> list)
```

Comment:
```suggestion
                    Span<IntPtr> fieldHandles, RuntimeType declaringType, ref ListBuilder<RuntimeFieldInfo> list)
```
`pp` prefix does not make sense for Span. Also, can it be `ReadOnlySpan` (Span->ReadOnlySpan casts cost nothing in typical code)?

Reply:
Yes, `ReadOnlySpan` works and updated parameter name.